### PR TITLE
Categorize by both category and tags

### LINKED ISSUE #515: Leverage hugo taxonomies to logically group content
Taxonomies in hugo helps to define logical relationships between the content. More infor - https://gohugo.io/content-management/taxonomies/

[AB#7127](https://dev.azure.com/azure-octo/e61041b4-555f-47ae-95b2-4f8ab480ea57/_workitems/edit/7127)

### PR DESCRIPTION
**Pull request by Reshrahim**
Thank you for helping make the Radius documentation better!

**Please follow this checklist before submitting:**

- [ ] [Read the contribution guide](https://docs.radapp.dev/contributing/docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

This PR covers the next iteration of leveraging taxonomies in our docs

1. Added categories to determine the type of document (Quickstart/Reference-app/How-to). Separate iteration can be done to cover other types of docs (Concepts doc, schema doc etc)

image

2. Added shortcode to categorize by the type of document (Category) and also by content metadata (Tags)

- Categorize quickstarts by tags 
  image

- Categorize reference-apps by tags 
- image

-  Categorize all types of docs by a tag 
- image

## Issue reference
#515

### CODE REVIEW

**Comment by AaronCrawfis** [comment]
This is AWESOME!

As part of this, could we update the docs contribution guidelines with definitions of what is a quickstart, and what is a how-to guide? I'm wondering if we might find they are pretty similar and we could maybe combine them 🤔. But not sure. We might also find they serve separate purposes.

**Comment by Reshrahim** [comment]
> This is AWESOME!
> 
> As part of this, could we update the docs contribution guidelines with definitions of what is a quickstart, and what is a how-to guide? I'm wondering if we might find they are pretty similar and we could maybe combine them 🤔. But not sure. We might also find they serve separate purposes.

Good point to clarify the definition of Quickstart and How-to. IMO, Quickstarts are guides to quickly learn and understand something in 3-5 mins while How-to's are guides that can be elaborate, cover advanced topics, and help the reader understand the nuances and details of a particular feature/topic area. Quickstarts are a subset of how-to's but they can cater to a different purpose than how-to's on the depth of a feature/area

File: docs/content/author-apps/container/index.md [hunk 1]
```diff
@@ -26,3 +26,10 @@ Depending on the target environment and configuration, a container can be run on
 ## More information
 
 - [Container schema]({{< ref container-schema >}})
+
+## Other guides
+
+{{< categorizeby category="Quickstart" tag="containers" >}}
+{{< categorizeby category="How-to" tag="containers" >}}
+
```
Comment (AaronCrawfis) [inline docs/content/author-apps/container/index.md]: Extra new line

File: docs/content/contributing/contributing-docs.md [hunk 2]
```diff
@@ -533,6 +563,20 @@ By default, buttons are padded with new lines below the button. To remove these
 {{< button text="Previous" link="https://example.com" newline="false" >}}
 {{< button text="Next" link="https://example.com" >}}
 
+#### Categorize content
```
Comment (AaronCrawfis) [inline docs/content/contributing/contributing-docs.md:569]: Headers should have a newline underneath

File: docs/content/getting-started/quickstarts/_index.md [hunk 3]
```diff
@@ -7,4 +7,6 @@ weight: 200
 no_list: true
 ---
 
-{{< categorize >}}
+{{< categorizeby category="Quickstart">}}
+
```
Comment (AaronCrawfis) [inline docs/content/getting-started/quickstarts/_index.md:11]: Extra new line

File: docs/content/contributing/contributing-docs.md [hunk 4]
```diff
@@ -109,22 +109,42 @@ The docs pipeline uses [aspell](http://aspell.net/) to check for spelling mistak
...
 
 ### Contributing a new How-To guide
 
 - Sub directory naming - the directory name should be descriptive and if referring to specific component or concept should begin with the relevant name. Example *pubsub-namespaces*.
 - Do not assume the reader is using a specific environment unless the article itself is specific to an environment. This include OS (Windows/Linux/MacOS), deployment target (Kubernetes, IoT etc.) or programming language. If instructions vary between operating systems, provide guidance for all.
 - Include code/sample/config snippets that can be easily copied and pasted.
 - At the end of the article, provide the reader with related links and next steps (this can be other relevant "how-to", samples for reference or related concepts).
+- Set the `category` as `How to` in [Hugo front-matter](#front-matter).
+
+### Contributing a new Reference-app
```
Comment (AaronCrawfis) [inline docs/content/contributing/contributing-docs.md]:
```suggestion
### Contributing a new Reference-app

```

File: docs/content/contributing/contributing-docs.md [hunk 5]
```diff
@@ -149,6 +169,8 @@ title: "TITLE FOR THE PAGE"
 linkTitle: "SHORT TITLE FOR THE NAV BAR"
 weight: (number)
 description: "1+ SENTENCES DESCRIBING THE ARTICLE"
+categories: "TYPE OF THE DOCUMENT"
+tags: "METADATA ON THE DOCUMENT"
```
Comment (AaronCrawfis) [inline docs/content/contributing/contributing-docs.md:174]: What would it look like if a document had 2 or more tags?

Comment (AaronCrawfis) [reply docs/content/contributing/contributing-docs.md:174]: Ah see the note below, way ahead of me :)

File: docs/content/getting-started/reference-apps/aws-rds/index.md [hunk 6]
```diff
@@ -4,6 +4,8 @@ title: "Deploy an AWS RDS-backed WordPress site"
 linkTitle: "WordPress on AWS"
 description: "Learn about how to use Radius to deploy a WordPress site that uses an AWS RDS database"
 weight: 600
+categories: "Reference-app"
```
Comment (AaronCrawfis) [inline docs/content/getting-started/reference-apps/aws-rds/index.md]:
```suggestion
categories: "Reference application"
```

File: docs/content/contributing/contributing-docs.md [hunk 7]
```diff
@@ -161,18 +183,26 @@ title: "Service invocation overview"
...
-Front-matter should be completed with all fields including type, title, linkTitle, weight, and description.
+Front-matter should be completed with all fields including type, title, linkTitle, weight, description, categories and tags.
 
 - `title` should be 1 sentence, no period at the end
 - `linkTitle` should be 1-3 words, with the exception of How-to at the front.
 - `description` should be 1-2 sentences on what the reader will learn, accomplish, or do in this doc.
+- `categories` should be one of the following:
+    - Concept
+    - Quickstart
+    - Reference app
```
Comment (AaronCrawfis) [inline docs/content/contributing/contributing-docs.md]:
```suggestion
    - Reference application
```

File: docs/content/contributing/contributing-docs.md [hunk 8]
```diff
@@ -161,18 +183,26 @@ title: "Service invocation overview"
...
+    - How to
```
Comment (AaronCrawfis) [inline docs/content/contributing/contributing-docs.md]:
```suggestion
    - How-To
```

File: docs/content/getting-started/reference-apps/aws-s3/_index.md [hunk 9]
```diff
@@ -4,6 +4,8 @@ title: "Using AWS S3 with Radius"
 linkTitle: "Using AWS S3"
 description: "Learn about how to model AWS S3 resources in Bicep and Radius"
 weight: 500
+categories: "Reference-app"
```
Comment (AaronCrawfis) [inline docs/content/getting-started/reference-apps/aws-s3/_index.md]:
```suggestion
categories: "Reference application"
```

File: docs/content/getting-started/reference-apps/container-app-store/index.md
Comment (AaronCrawfis) [inline docs/content/getting-started/reference-apps/container-app-store/index.md:1]: This file was removed in a previous PR. I'd make sure you've pulled the latest 'edge' branch. If so I'd delete this directory

File: docs/content/getting-started/reference-apps/eshop-dapr/_index.md [hunk 10]
```diff
@@ -4,6 +4,8 @@ title: "eShop on Dapr sample"
 linkTitle: "eShop on Dapr"
 description: "Learn how to model and deploy a fully portable application with Dapr and Project Radius"
 weight: 150
+categories: "Reference-app"
```
Comment (AaronCrawfis) [inline docs/content/getting-started/reference-apps/eshop-dapr/_index.md]:
```suggestion
categories: "Reference application"
```

File: docs/content/getting-started/reference-apps/eshop/_index.md [hunk 11]
```diff
@@ -4,6 +4,8 @@ title: "eShop on containers sample"
 linkTitle: "eShop"
 description: "Learn how to model and deploy a production-level application with Radius"
 weight: 100
+categories: "Reference-app"
```
Comment (AaronCrawfis) [inline docs/content/getting-started/reference-apps/eshop/_index.md]:
```suggestion
categories: "Reference application"
```

Comment (AaronCrawfis) [inline docs/content/getting-started/reference-apps/eshop/_index.md]:
```suggestion
categories: "Reference application"
```

File: docs/content/operations/platforms/kubernetes/kubernetes-metadata/index.md [hunk 12]
```diff
@@ -4,6 +4,8 @@ title: "Set Kubernetes metadata"
 linkTitle: "Kubernetes metadata"
 description: "Learn how to configure Kubernetes labels and annotations for generated objects"
 weight: 300
+categories: "How-to"
```
Comment (AaronCrawfis) [inline docs/content/operations/platforms/kubernetes/kubernetes-metadata/index.md]:
```suggestion
categories: "How-To"
```

**Review by AaronCrawfis — CHANGES_REQUESTED** [review]
Looking good! I'm thinking for the categories, "Reference application" would be better without the - and with app spelled out.

And for How-To, we've been capitalizing both words in the docs themselves. We should probably be consistent in the metadata as well.

File: docs/content/getting-started/reference-apps/container-app-store/index.md
Comment (Reshrahim) [reply docs/content/getting-started/reference-apps/container-app-store/index.md:1]: removed it!

**Review by AaronCrawfis — APPROVED** [review]

**Pull request merged**